### DISCUSSION TITLE
How add route along with subroutes

### DISCUSSION THREAD

**Discussion by HardCoreQual** [RobinTail/express-zod-api#536] [Q&A] [answered]
for  example

```
export const routing: Routing = {
  files: filesEndpoint,
  files: { /// it's don't work because object key is duplicated
    ":type" : filesByTypeEndpoint
  },
};
```

**Comment by teaishealthy** [comment, marked as answer]
You would do it like this 
```js
export const routing: Routing = {
  files: {
    ":type": filesByTypeEndpoint,
    "": filesEndpoint
  },
```

**Comment by RobinTail** [comment]
@HardCoreQual , if by `:type` you mean the route parameter, then you follow the @teaishealthy 's advice.
Otherwise, please provide more clarifications.

Possible copy of https://github.com/RobinTail/express-zod-api/discussions/73

**Reply by HardCoreQual** [reply]
yes, I asked about case then exist current route, but also exist and subroutes

it also can be
```typescript
   const router = {
      file:  {
           "": getFile,
           meta: getFileMata,
      },
   }
```
can be need add example for this case in readme, or it already exist and I missing it ?

**Reply by RobinTail** [reply]
https://github.com/RobinTail/express-zod-api#route-path-params , @HardCoreQual

**Reply by HardCoreQual** [reply]
yes, in readme hasn't case `"": endpoint`

**Reply by RobinTail** [reply]
thank you for noticing this, @HardCoreQual .
I've just fixed it.
